internal/pkg/skuba/upgrade/node/versions.go

```diff
@@ -152,16 +161,32 @@ func controlPlaneUpdateStatusWithAvailableVersions(currentClusterVersion *versio
 	}, nil
 }
 
-func workerUpdateStatus(currentClusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node) (NodeVersionInfoUpdate, error) {
-	return workerUpdateStatusWithAvailableVersions(currentClusterVersion, allNodesVersioningInfo, node, kubernetes.StaticVersionInquirer{})
+func workerUpdateStatus(clusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node) (NodeVersionInfoUpdate, error) {
+	return workerUpdateStatusWithAvailableVersions(clusterVersion, allNodesVersioningInfo, node, kubernetes.StaticVersionInquirer{})
 }
 
-func workerUpdateStatusWithAvailableVersions(currentClusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node, versionInquirer kubernetes.VersionInquirer) (NodeVersionInfoUpdate, error) {
+func workerUpdateStatusWithAvailableVersions(clusterVersion *version.Version, allNodesVersioningInfo kubernetes.NodeVersionInfoMap, node *v1.Node, versionInquirer kubernetes.VersionInquirer) (NodeVersionInfoUpdate, error) {
+	// Checking worker nodes for updates is a bit different than checking a control plane node.
+	// It can be that an upgrade has already been started on the control plane
+	// or that all nodes are still on the same version (no upgrade started yet).
+	// First we need to check if an upgrade has aready been started
+	// This can be determined by kubernetes.AllNodesMatchClusterVersion(allNodesVersioningInfo, clusterVersion)
+	allNodesMatchCurrentClusterVersion, err := kubernetes.AllNodesMatchClusterVersionWithVersioningInfo(allNodesVersioningInfo, clusterVersion)
+	if err != nil {
+		return NodeVersionInfoUpdate{}, err
+	}
 	// Check that all control plane nodes have at least the current cluster version we plan to
 	// upgrade this worker node to. If not, they need to be fully upgraded first
-	controlPlanesMatchVersion := kubernetes.AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, currentClusterVersion)
-	if !controlPlanesMatchVersion {
-		return NodeVersionInfoUpdate{}, errors.New("at least one control plane does not tolerate the current cluster version")
+	controlPlanesMatchVersion := kubernetes.AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, clusterVersion)
+	// Check if there is a newer version
+	versionCompare, err := clusterVersion.Compare(kubernetes.LatestVersion().String())
+	if err != nil {
+		return NodeVersionInfoUpdate{}, err
+	}
+	if versionCompare != 0 && (allNodesMatchCurrentClusterVersion || !controlPlanesMatchVersion) {
```

Comment:
If we are checking for newer let's avoid `!=` and use `<` `>` comparisons better, just to be sure.